### PR TITLE
Some HSI improvements

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -5040,11 +5040,23 @@ fu_engine_get_host_machine_id (FuEngine *self)
...
 {
+	gboolean disabled_plugins = FALSE;
+	GPtrArray *blacklist = fu_config_get_blacklist_plugins (self->config);
 	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.fwupd.Hsi.Plugins");
 	fwupd_security_attr_set_name (attr, "fwupd plugins");
 	fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
+	for (guint i = 0; i < blacklist->len; i++) {
+		const gchar *name_tmp = g_ptr_array_index (blacklist, i);
+		if (g_strcmp0 (name_tmp, "test") != 0 &&
+		    g_strcmp0 (name_tmp, "invalid") != 0) {
```
Comment (hughsie) [inline src/fu-engine.c:5051]: Not sure about the hardcoding, but for these two it's probable fine.

File: src/fu-engine.c [hunk 2]
```diff
@@ -5040,11 +5040,23 @@ fu_engine_get_host_machine_id (FuEngine *self)
 static void
 fu_engine_add_security_attrs_tainted (FuEngine *self, FuSecurityAttrs *attrs)
 {
+	gboolean disabled_plugins = FALSE;
+	GPtrArray *blacklist = fu_config_get_blacklist_plugins (self->config);
 	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.fwupd.Hsi.Plugins");
```
Comment (hughsie) [inline src/fu-engine.c:5045]: Ohh, this needs to be `g_autoptr()` -- my bad. Can you fix that up pls.

File: src/fu-util-common.c [hunk 3]
```diff
@@ -1610,9 +1611,21 @@ fu_util_security_attrs_to_string (GPtrArray *attrs)
 				FwupdSecurityAttr *attr = g_ptr_array_index (attrs, i);
 				if (!fwupd_security_attr_has_flag (attr, hpi_suffixes[j]))
 					continue;
+				if (!help &&
```
Comment (hughsie) [inline src/fu-util-common.c:1614]: maybe lose this, micro-optimization.

File: src/fu-util-common.c [hunk 4]
```diff
@@ -1610,9 +1611,21 @@ fu_util_security_attrs_to_string (GPtrArray *attrs)
...
+				    fwupd_security_attr_has_flag (attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE) &&
+				    !fwupd_security_attr_has_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS))
```
Comment (hughsie) [inline src/fu-util-common.c:1616]: woah, that's quite a high bar -- if you fail a HSI:5 check and you drop down to HSI:4 that's really still very high and I don't think we should show the notification. I'd much rather have two pages, one for runtime issues, and one if the HSI number is < 2 or something.

File: src/fu-util-common.c [hunk 5]
```diff
@@ -1610,9 +1611,21 @@ fu_util_security_attrs_to_string (GPtrArray *attrs)
...
 				fu_security_attr_append_str (attr, str);
 			}
 		}
 	}
+
+	if (help) {
+		/* TRANSLATORS:  this is instructions on how to improve the HSI suffix */
+		g_string_append_printf (str, "\n%s: %s",
+					_("Your system has runtime issues. To improve your security visit"),
+					"https://github.com/fwupd/fwupd/wiki/Low-host-security-ID");
```
Comment (hughsie) [inline src/fu-util-common.c:1627]: I don't think translators like having URLs - maybe build the string using `g_string_new()` and use %s for the URL.

File: src/fu-util-common.c (hunk 4 above)
Comment (superm1) [reply src/fu-util-common.c:1616]: OK i'll rename that page and make it runtime focused (as that's what the PR is).

File: src/fu-engine.c (hunk 2 above)
Comment (superm1) [reply src/fu-engine.c:5045]: sure, same problem elsewhere it seems too.

File: src/fu-engine.c (hunk 1 above)
Comment (superm1) [reply src/fu-engine.c:5051]: The other option is to only include these when plugin_dummy is compiled, but I think for distro purposes that's all the time, so I think I would rather keep them hardcoded.

File: src/fu-util-common.c (hunk 5 above)
Comment (superm1) [reply src/fu-util-common.c]: I adjusted it to model off of what was done in `fu-util.c` for a URL.

File: src/fu-util-common.c (hunk 3 above)
Comment (superm1) [reply src/fu-util-common.c]: fixed

File: src/fu-util-common.c [hunk 6]
```diff
@@ -1610,9 +1616,26 @@ fu_util_security_attrs_to_string (GPtrArray *attrs)
...
 	}
+
+	if (low_help) {
+		/* TRANSLATORS:  this is instructions on how to improve the HSI security level */
+		g_string_append_printf (str, "\n%s %s",
+					_("Your system has a low HSI security level. Visit this URL for more information: "),
+					"https://github.com/fwupd/fwupd/wiki/Low-host-security-level");
+	}
+	if (runtime_help) {
+		/* TRANSLATORS:  this is instructions on how to improve the HSI suffix */
```
Comment (hughsie) [inline src/fu-util-common.c:1634]: frustratingly, the TRANSLATORS line has to be the line below, it only gets found if it's one logical line above the _()

Comment (superm1) [reply src/fu-util-common.c]: wow, that's annoying.  OK fixed.

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c:5051]: Ya, I think two is fine.

**Pull request merged**